cli/install/install.go

```diff
@@ -0,0 +1,747 @@
+		return false
+	} else {
+		return true
+	}
+}
+
+// executeCommand executes program with given args in verbose or quiet mode.
+func executeCommand(program string, isVerbose bool, logFile *os.File, path string,
+	args ...string) error {
+	cmd := exec.Command(program, args...)
+	if isVerbose {
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+	} else {
+		cmd.Stdout = logFile
+		cmd.Stderr = logFile
+	}
+	if path == "" {
+		path, _ = os.Getwd()
+	}
+	cmd.Dir = path
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+	err = cmd.Wait()
+	return err
+}
+
+// changeSymlink changes symlink.
+func changeSymlink(version string, dst string, program string) error {
+	path := filepath.Join(dst, program)
+	os.Remove(path)
+	err := os.Symlink(version, path)
+	return err
+}
+
+// downloadRepo downloads git repository.
+func downloadRepo(tag string, logFile *os.File, verbose bool, dst string,
+	repoLink string) error {
+	var err error
+	if tag == "master" {
+		err = executeCommand("git", verbose, logFile, dst, "clone",
+			"--depth=1", "-j", "18", repoLink,
+			"--recursive", dst)
+	} else {
+		err = executeCommand("git", verbose, logFile, dst, "clone", "-b", tag,
+			"--depth=1", "-j", "18", repoLink,
+			"--recursive", dst)
+	}
+
+	return err
+}
+
+// installTt installs selected version of tt.
+func installTt(version string, dst string, flags InstallationFlag, local string) error {
+
+	// Check binary directory.
+	if dst == "" {
+		return fmt.Errorf("Cannot procces bindir.")
+	}
+	logFile, _ := ioutil.TempFile("", "tarantool_install")
+
+	// Check if that version is already installed.
+	log.Warnf("Checking existing...")
+	if checkExisting(version, dst) && !flags.Reinstall {
+		log.Warnf("That version already exists, changing symlink...")
+		err := changeSymlink(version, dst, "tt")
+		log.Warnf("Done")
+		return err
+	}
+
+	// Check tt dependencies.
+	log.Warnf("Checking dependencies...")
+	if !checkProgrammDependencies("tt") && !flags.Force {
+		return nil
+	}
+
+	path, err := os.MkdirTemp("", "tt_install")
+	os.Chmod(path, 0777)
+	if err != nil {
+		return err
+	}
+	if !flags.Noclean {
+		defer os.RemoveAll(path)
+	}
+
+	// Download tt.
+	if flags.Local {
+		if checkExisting("tt", local) {
+			log.Warnf("Local files found, installing from them...")
+			localPath, _ := util.JoinAbspath(local, "tt")
+			err = util.CopyDir(localPath, path)
+			if err != nil {
+				logs, _ := os.ReadFile(logFile.Name())
+				os.Stdout.Write(logs)
+				return err
+			}
+			executeCommand("git", flags.Verbose, logFile, path, "checkout", "master")
+		} else {
+			return fmt.Errorf("Can't find distfiles directory.")
+		}
+	} else {
+		log.Warnf("Downloading tt...")
+		err = downloadRepo("master", logFile, flags.Verbose, path, TTLink)
+	}
+
+	if err != nil {
+		logs, _ := os.ReadFile(logFile.Name())
+		os.Stdout.Write(logs)
+		return err
+	}
+	// Build tt.
+	log.Warnf("Building tt...")
+	err = executeCommand("mage", flags.Verbose, logFile, path,
```

Comment:
This looks very strange: program name "mage" and "build" args are separated. I offer to use exec.Cmd struct to store everything related to command execution.
Consider using execute.go file from cartridge.